pubsub/pubsub.rb

```diff
@@ -0,0 +1,105 @@
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "sinatra"
+require "gcloud"
+require "logger"
+require "json"
+require "base64"
+
+configure do
+  set :port, 8080
+
+  set :topic_name, "my_topic"
+  set :sub_name, "my_topic_sub"
+  set :sub2_name, "my_topic_sub_push"
+  set :project, "my_project_id"
+  set :project, ENV["GCP_PROJECT_ID"] unless ENV["GCP_PROJECT_ID"].nil?
+
+  gcloud = Gcloud.new settings.project
+  pubsub = gcloud.pubsub
+
+  # Get topic, create if not created
+  topic = pubsub.topic settings.topic_name, autocreate: true
+
+  sub = topic.get_subscription settings.sub_name
+  topic.subscribe settings.sub_name if sub.nil?
+
+  sub2 = topic.get_subscription settings.sub2_name
+  topic.subscribe(
+    settings.sub2_name,
+    endpoint: "https://#{settings.project}.appspot.com/push") if sub2.nil?
+
+  $logger = Logger.new(STDERR)
+  $logger.level = Logger::DEBUG
+  $logger.debug "All set up."
+end
+
+get "/" do
+  "Hello World!"
+end
+
+get "/_ah/health" do
+  "Hello World!"
+end
+
+post "/pub" do
+  request.body.rewind
+  gcloud = Gcloud.new settings.project
+  topic = gcloud.pubsub.topic settings.topic_name
```

Comment:
In these code samples, let's make `pubsub` its own variable to be explicit and to follow the coding style demonstrated in the gcloud-ruby [topic samples](http://googlecloudplatform.github.io/gcloud-ruby/docs/v0.6.1/Gcloud/Pubsub/Topic.html).

```ruby
gcloud = Gcloud.new
pubsub = gcloud.pubsub
topic = pubsub.topic "topic_name"
```